### DISCUSSION TITLE
[docs] Plugins can set up and maintain state doesn't work

### DISCUSSION THREAD

**Discussion by fmasion** [tauri-apps/tauri#7043] [Frequently Asked]
Hi, 

In the documentation, page [https://tauri.app/fr/v1/guides/features/plugin/](https://tauri.app/fr/v1/guides/features/plugin/),  there is an example teaching how to create a plugin, using the plugin::Builder, **that manages state** :

Here it is :

`pub fn init<R: Runtime>() -> TauriPlugin<R> {
  Builder::new("awesome")
    .invoke_handler(tauri::generate_handler![do_something])
    .setup(|app_handle| {
      // setup plugin specific state here
      app_handle.manage(MyState::default());
      Ok(())
    })
    .build()
}`

Sadly, `app_handle` doesn't have any `manage()` method anymore...

I found out that I can manage my plugin state in the app builder directly...
It works, but the app should not be aware of the plugin's state (imho). So I assume it should be a better way to do it.

Thanx for help

Fred

**Comment by FabianLars** [comment]
Did you add `use tauri::Manager` at the top of your rs file?

**Reply by fmasion** [reply]
my bad sorry
thank for your help

**Comment by wuhaihai1** [comment]
OK

**Comment by jose8109** [comment]
michas graxias por la ayuda,  como sabran , he estado teniendo problemas , y von gisto ayudare soy nuevo y apredere rapido, gracias, me estaban destrosando todo

**Comment by jose8109** [comment]
ayuda

**Comment by jose8109** [comment]
Help me están vaciando la billeteras virtual
